internal/flusher.go

```diff
@@ -0,0 +1,55 @@
+package internal
+
+import (
+	"log"
+	"time"
+)
+
+type BackgroundFlusher interface {
+	Start()
+	Stop()
+}
+
+type backgroundFlusher struct {
+	ticker   *time.Ticker
+	interval time.Duration
+	handler  LineHandler
+	stop     chan struct{}
+}
+
+func NewBackgroundFlusher(interval time.Duration, handler LineHandler) BackgroundFlusher {
+	return &backgroundFlusher{
+		interval: interval,
+		handler:  handler,
+		stop:     make(chan struct{}),
+	}
+}
+
+func (f *backgroundFlusher) Start() {
+	format := f.handler.(*RealLineHandler).Format
```

Comment:
nit: make this part of the interface so that it doesn't require casting here

Reply:
Like this